tuplex/core/include/logical/TakeOperator.h

```diff
@@ -37,8 +38,9 @@ namespace tuplex {
 
         bool good() const override;
 
-        int64_t limit() { return _limit; }
+        int64_t limit() { return _limitTop; }
```

Comment:
make function const, also add a check that when limit() is used, then bottomLimit should be 0 (assert!)

add new function topLimit() and bottomLimit() (make const functions).